src/main/java/org/folio/service/transactions/batch/BatchPaymentCreditService.java

```diff
@@ -73,23 +74,30 @@ private void updateEncumbranceToCancelTransaction(Transaction transaction, Trans
     // NOTE: we can't unrelease the encumbrance automatically because the payment/credit does not say if the encumbrance
     // was released automatically or not (as opposed to a pending payment).
     double expended = encumbrance.getEncumbrance().getAmountExpended();
+    double credited = encumbrance.getEncumbrance().getAmountCredited();
     double amount = encumbrance.getAmount();
     if (transaction.getTransactionType() == PAYMENT) {
       expended = subtractMoney(expended, transaction.getAmount(), currency);
       amount = sumMoney(amount, transaction.getAmount(), currency);
     } else {
-      expended = sumMoney(expended, transaction.getAmount(), currency);
+      if (transaction.getTransactionType() == CREDIT) {
```

Comment:
This is within the else for "`if (transaction.getTransactionType() == PAYMENT) {`" so we are already in the `CREDIT` case. `amountExpended` should be changed in the case of a `PAYMENT`, and `amountCredited` should be changed in the case of a `CREDIT`.

Reply:
I was thinking of something like that:
```
    if (transaction.getTransactionType() == PAYMENT) {
      expended = subtractMoney(expended, transaction.getAmount(), currency);
      amount = sumMoney(amount, transaction.getAmount(), currency);
    } else {
      credited = subtractMoney(credited, transaction.getAmount(), currency);
      if (Encumbrance.Status.RELEASED != encumbrance.getEncumbrance().getStatus()) {
        amount = subtractMoney(amount, transaction.getAmount(), currency);
      }
    }
```
There is something strange about it though: it is not symetrical between payment and credit, with that check for a released encumbrance. But this is an issue with the initial code. It should probably be investigated in a separate ticket.

Reply:
changed based on your comment in other places as well, thanks for notice it

Reply:
I am going to create that ticket.